### PR TITLE
Update nrf-802154 revision

### PR DESCRIPTION
**Pull request by jciupis**

### CODE REVIEW

**Review by ankuns — APPROVED** [review]

**Review by rugeGerritsen — APPROVED** [review]
It would be nice if we could stop using magic numbers for EGU channel allocation

**Review by pkwiek — APPROVED** [review]

**Comment by kkasperczyk-no** [comment]
Please rebase after merging this PR: https://github.com/nrfconnect/sdk-nrf/pull/6433 to fix Matter weather station app CI issues.

File: west.yml [hunk 1]
```diff
@@ -104,7 +104,7 @@ manifest:
     - name: nrfxlib
       repo-path: sdk-nrfxlib
       path: nrfxlib
-      revision: eb80e86ffc75354132ade2da89b09cd26bc4d9b7
+      revision: pull/587/head
```
Comment (rlubos) [inline west.yml:107]: `d188f23ee7677760a3464209aabd77bba6d9a75f`

File: west.yml [hunk 2]
```diff
@@ -54,7 +54,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: v2.7.0-ncs1
+      revision: pull/682/head
```
Comment (rlubos) [inline west.yml:57]: `0e85a89ccb6039a8736a7c9b0913c860fb047722`

**Review by rlubos — APPROVED** [review]

**Pull request merged**